rust/noosphere-core/src/view/sphere.rs

```diff
@@ -805,13 +805,13 @@ impl<S: BlockStore> Sphere<S> {
                 ));
             }
 
+            let ucan_issuer = chain_link.ucan().issuer();
+
             if let Some(revocation) = revocations.get(&link).await? {
-                // NOTE: The implication here is that only the sphere itself can
-                // issue revocations The follow-on implicationis that a user
-                // must provide the sphere mnemonic in order to issue a
-                // revocation
-                if revocation.iss != sphere_identity {
-                    warn!("Revocation for {} had an invalid issuer; expected {}, but found {}; skipping...", link, sphere_identity, revocation.iss);
+                // NOTE: The implication here is that only the sphere itself or
+                // the direct issuer may issue revocations
+                if revocation.iss != sphere_identity || revocation.iss != ucan_issuer {
```

Comment:
What's the significance of the ucan issuer? Is that an identity that's allowed to impersonate the sphere_identity?

Reply:
[Revocation structures](https://github.com/ucan-wg/spec#66-revocation) are related to UCAN, but their own unique structure. In UCAN parlance, the issuer is the signer of the UCAN token (or revocation). Here, we check both that the revocations either came this sphere, or whoever issued the revocation in the first place